src/graphics/particle/MagicFlare.cpp

```diff
@@ -188,8 +188,7 @@ void AddFlare(const Vec2f & pos, float sm, short typ, Entity * io, bool bookDraw
 		flare.flags = 0;
 	}
 
-	flare.pos.x = pos.x - Random::getf(0.f, 4.f);
-	flare.pos.y = pos.y - Random::getf(0.f, 4.f);
+	flare.pos = pos - Random::getf(0.f, 4.f);
```

Comment:
This is not the same - in the new code the random x and y components are always the same.